### PR TITLE
#184637142 feat: User update password

### PR DESCRIPTION
**Pull request by mjeanbosco19**
### What does this PR do?

- User should be able to update their password

### Description of Task to be completed?

- [x] The user might want to change the password in case they want to. this is different from resetting the password they have forgot.

1. GIVEN user is validly registered and account is validated.

2. THEN users new password should be hashed and updated, or rejected due to the user not remembering old password or new password and confirm password fields not matching. 

### How should this be manually tested?

**Installation**
**1. Clone the repository by running** 
```
git clone https://github.com/atlp-rwanda/destructors-ec-bn.git
```
**2. Navigate to the project directory by running cd destructors-ec-bn.**
```
cd destructors-ec-bn
```
```
cd ft-update-password-184637142
```
**3. Install the necessary dependencies by running npm install.**
```
npm install
```
**4. Run the development server by running npm run dev. This will start the server using Nodemon, which will automatically reload the server when changes are made to the code.**
```
npm run dev
```
**5. Before update password roles login and save token in headers** 
```
POST: /api/v1/users/login
```
**6. Update your password on below endpoint** 
```
PATCH: /api/v1/users/update-password'
```
**7. Run the tests by running npm run test** 
```
npm run test
```
### Any background context you want to provide?
when running this on local server put into consideration the below information 👇 

- [x] add the your `login token` in headers with a key of `Authorization` 

### What are the relevant pivotal tracker/Trello stories?

- Tracker: 184637142

### Screenshots (if appropriate)

- none

### Questions:

- none

### CODE REVIEW

**Review by katros1 — APPROVED** [review]

File: src/routes/api/updatePassword.routes.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import express from 'express';
```
Comment (leandreAlly) [inline src/routes/api/updatePassword.routes.js]: this routes must be removed, and appended to user.routes.js

File: src/routes/index.js [hunk 2]
```diff
@@ -1,11 +1,13 @@
 import express from 'express';
 import user from './api/user.routes.js';
 import product from './api/products.routes.js';
+import updateUserPassword from './api/updatePassword.routes';
 
 
 const routes = express.Router();
 routes.use('/users', user);
 routes.use('/products', product);
+routes.use('/users', updateUserPassword);
```
Comment (leandreAlly) [inline src/routes/index.js]: This is duplication of 
routes.use('/users', updateUserPassword);
where this must be appended to routes.use('/users', user);

File: src/controllers/user.controller.js [hunk 3]
```diff
@@ -181,5 +182,44 @@ try {
...
+    const userId = req.user.id;
+    const user = await findUserById(userId);
+
+    const { currentPassword, newPassword, confirmPassword } = req.body;
+    // Check current password
+    const comparePassword = await BcryptUtil.compare(
+      currentPassword,
+      user.password
+    );
+    console.log(comparePassword);
```
Comment (ba-lambert) [inline src/controllers/user.controller.js]: It would be better if you remove these consoles

File: src/routes/api/updatePassword.routes.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import { updatePassword } from '../../controllers/user.controller';
+import updatePasswordValidation from '../../ validations/updatePassword.validation';
+import extractToken from '../../middlewares/checkUserWithToken';
+
+const router = express.Router();
+
+router.patch(
```
Comment (ba-lambert) [inline src/routes/api/updatePassword.routes.js]: Your route should be in the users route

**Review by leandreAlly — CHANGES_REQUESTED** [review]

**Review by ba-lambert — APPROVED** [review]
I have seen a problem on routes which should be in users route and also it would be if you remove the consoles

File: src/controllers/user.controller.js [hunk 5]
```diff
@@ -181,5 +182,44 @@ try {
...
+
+    user.password = hashedPassword;
+    await user.save();
+
+    res.status(200).json({
+      success: true,
+      data: 'Password updated successfully',
+    });
+  } catch (err) {
+    next(err);
```
Comment (leandreAlly) [inline src/controllers/user.controller.js:219]: why have you called this next()?

**Review by nkurunziza1 — APPROVED** [review]

File: src/controllers/user.controller.js (hunk 5 above)
Comment (mjeanbosco19) [reply src/controllers/user.controller.js:219]: > why have you called this next()?

The purpose of next(err) is to pass the error to the next middleware function in the stack.

**Pull request merged**